### DISCUSSION TITLE
Out of Range Error for bigNumberify on testnet BUT no error on devnet

### DISCUSSION THREAD

**Discussion by asolpshinning** [reach-sh/reach-lang#1165] [Q&A] [answered]
So the API below is for claiming/receiving some amount of non-network tokens (BT) from the smart contract. This API works well on devnet even when trying to claim 990 million backing tokens. But on testnet, I am not getting any error for small numbers in tens of thousands, but getting the reach error below once I am trying to claim/receive millions of tokens from the contract. What could be the reason ?

I got this error when trying to claim 90 million tokens:

`There is an error:  Error: cBT errored with Error: bigNumberify: 58500000000000000000 out of range [0, 18446744073709551615] `

![image](https://user-images.githubusercontent.com/57355937/174438042-8fcc98dd-088b-4e11-a6b6-b594a3b35808.png)

**Comment by jeapostrophe** [comment, marked as answer]
There's not enough context to help. How is it being called?

**Reply by asolpshinning** [reply]
Thanks @jeapostrophe. Here is the link to the [full code](https://github.com/cooperativ-labs/share-manager-contract-algorand/blob/main/index.rsh).

It seems I am not longer getting the error after using `UInt256` for variables used in computations in the APIs, and re-odering the math operation by doing division first before multiplication. I am not exactly sure what helped. It would be great to know what it is, if the information I have provided here helps.

However, to answer your question, `cBT` is being called like I usually call APIs See below: 

![image](https://user-images.githubusercontent.com/57355937/174490521-339994a1-48b5-4317-978e-89f19a8d8254.png)

**Reply by jeapostrophe** [reply]
It sounds like the values just overflow and Reach is detecting that before sending it to the network. When you switched to UInt256, there was more space available
